Tests/Tests/UserInfoProviderSpec.swift

```diff
@@ -157,6 +157,106 @@ class UserInfoProviderSpec: QuickSpec {
                     expect(userInfoProviderA == userInfoProviderB).to(beTrue())
                 }
             }
+
+            context("when checking inequality") {
+                var userInfoProviderA: UserInfoProviderMock!
+                var userInfoProviderB: UserInfoProviderMock!
+
+                beforeEach {
+                    userInfoProviderA = UserInfoProviderMock()
+                    userInfoProviderB = UserInfoProviderMock()
+                }
+
+                it("will return false for nil object") {
+                    userInfoProviderA = nil
+                    userInfoProviderB = nil
+                    expect(userInfoProviderA != userInfoProviderB).to(beFalse())
+                }
+
+                it("will return false for nil and empty objects") {
+                    userInfoProviderA = nil
+                    expect(userInfoProviderA != userInfoProviderB).to(beFalse())
```

Comment:
so empty object and nil are treated as equal? 🤔 It doesn't sound correct to me

Reply:
Yes, I think because both of the `userIds ` is empty it's equal.
In the UserInfoProvider | L.81 you made a comment:
**// null and empty values are treated the same**
So, it should be correct, right?

Reply:
Ahh ok. That makes sense. I remember now 😅. Nil and empty user are treated as the same to improve 'logout' state detection. The test is absolutely correct.